File: 1894-merge-strings-alternately/merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        int w1=0;
        
        StringBuilder s = new StringBuilder();
        while(w1< word1.length() || w1< word2.length()){
            if(w1<word1.length()){
                s.append(word1.charAt(w1));
            }
            if(w1<word2.length()){
                s.append(word2.charAt(w1));
            }
            
            w1++;
        }

        
        return s.toString();
    }

}

File: 1894-merge-strings-alternately/merge-strings-alternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        result = []
        x=0
        while x < len(word1) or x < len(word2):
            if x < len(word1):
                result.append(word1[x])
            if x < len(word2):
                result.append(word2[x])
            x+=1
        
        return "".join(result)
            

        